mark.c

```diff
@@ -619,7 +619,11 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
 {
   signed_word credit = HBLKSIZE;  /* Remaining credit for marking work. */
   ptr_t current_p;      /* Pointer to current candidate ptr.            */
+# if !defined(__CHERI_PURE_CAPABILITY__)
   word current;         /* Candidate pointer.                           */
+# else
+  void **current;       /* Candidate pointer.                           */
```

Comment:
I've a feeling that having some conditional `typedef` for `word`-vs-`void**` might simplify some later code blocks and remove some duplication. I'm not certain, but I think some blocks are duplicated only (or primarily) to change these types.

Reply:
The `word`  macro is being used in a lot of places in the code base  - so I did not want to replace that everywhere.  I think this is minimally invasive.

Reply:
I don't mean replace the definition of `word`, I mean just for this file/function/context, make something like:

```c++
#ifdef __CHERI_PURE_CAPABILITY__
typedef word mark_word;    // Maybe pick a better name!
#else
typedef void** mark_word;
#endif
```

... then use `mark_word` in the body instead of duplicating.

It might not be worthwhile (e.g. if other changes lead to duplication anyway).

Reply:
I have removed the code duplication in commit [c1adac6](https://github.com/capablevms/bdwgc/pull/13/commits/c1adac6865660965561975acc3587513d3299e8e).  If this is OK, I will leave it as it is. If not I can implement  a new typdef @jacobbramley

Reply:
Agreed, it seems fine now. I'll resolve this, thanks :-)